bool detected
int32 center_x
int32 center_y
float64 side_diff
float64 area_ratio